mersenne-31/src/mersenne_31.rs

```diff
@@ -32,16 +32,30 @@ pub struct Mersenne31 {
 }
 
 impl Mersenne31 {
+    /// Convert a u32 element into a Mersenne31 element.
+    ///
+    /// # Safety
+    /// The element must lie in the range: `[0, 2^31 - 1]`.
     #[inline]
-    pub const fn new(value: u32) -> Self {
+    pub(crate) const fn new(value: u32) -> Self {
         debug_assert!((value >> 31) == 0);
         Self { value }
     }
 
+    /// Convert a u32 element into a Mersenne31 element.
+    ///
+    /// # Panics
+    /// This will panic if the element does not lie in the range: `[0, 2^31 - 1]`.
+    #[inline]
+    pub const fn new_checked(value: u32) -> Self {
+        assert!((value >> 31) == 0);
+        Self { value }
+    }
```

Comment:
Hmm, some people would prefer such a method to return an `Option`:
```suggestion
    pub const fn new_checked(value: u32) -> Self {
        ((value >> 31) == 0).then(|| Self { value })
    }
```
Up to you.

Reply:
Yeah I think about that but didn't realize that `.unwrap()` works in constant environments. (Otherwise this would be a little annoying as you'd have `Option<Mersenne31>`'s floating around in all your constants).

Given that though, returning `Option` seems good (though for silly reasons we can't use then as that isn't const)

Reply:
Oh, good catch. Clearly I didn't check my code before posting it. :sweat_smile: